app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/DownloadAttachmentProgressDialog.kt

```diff
@@ -76,8 +79,14 @@ class DownloadAttachmentProgressDialog : DialogFragment() {
     }
 
     private fun popBackStackWithError() {
-        val title = if (mainViewModel.isInternetAvailable.value == true) R.string.anErrorHasOccurred else R.string.noConnection
-        showSnackbar(title)
-        findNavController().popBackStack()
+        // We have to use the Flow instead of the LiveData here since the mainViewModel is not the same instance as the one in the
+        // MainActivity. So, instead of observing the LiveData in order to make it "active", we use the Flow.
```

Comment:
What LiveData are you talking about ? The one that you removed ?

Reply:
The previous LiveData we had before that modification. But I can remove that explanation if that's misleading.

Reply:
Yes you can.